src/core/secretsManager.ts

```diff
@@ -1,73 +1,338 @@
-import type { SecretStorage, Disposable } from "vscode";
+import { toSafeHost } from "../util";
 
-const SESSION_TOKEN_KEY = "sessionToken";
+import type { Memento, SecretStorage, Disposable } from "vscode";
 
-const LOGIN_STATE_KEY = "loginState";
+import type { TokenResponse, ClientRegistrationResponse } from "../oauth/types";
 
-export enum AuthAction {
-	LOGIN,
-	LOGOUT,
-	INVALID,
+import type { Deployment } from "./deployment";
+
+const SESSION_KEY_PREFIX = "coder.session.";
```

Comment:
Thinking out loud, right now we have `$key_deployment` for each type of secret but would it make sense to have one secret for all the keys?

So instead of:

```
coder.session.dev.coder.com = "foo"
coder.oauth.tokens.dev.coder.com = "foo"

```

we have something like:

```
coder.deployments.dev.coder.com = JSON.stringify({
  session: "foo",
  oauth: { tokens: "foo" },
})
```